hyperon_das_atomdb/database.py

```diff
@@ -245,8 +238,8 @@ def _build_link(
             if key not in reserved_parameters and not re.search(AtomDB.key_pattern, key)
         }
 
-        link_type = valid_params.get('type')
-        targets = valid_params.get('targets')
+        targets = link_params.get("targets")
+        link_type = link_params.get("type")
 
         if link_type is None or targets is None:
```

Comment:
`link_type` is being assigned and check in the `if` below but is not being used anywhere else.

Reply:
This seems correct. It's not used but the check raises an exception if it's absent. This is important because it assures that links going to MongoDB have the field properly set.